pkg/sfu/prober.go

```diff
@@ -125,12 +125,15 @@ type Prober struct {
 
 	clusterId atomic.Uint32
 
-	clustersMu    sync.RWMutex
-	clusters      deque.Deque
-	activeCluster *Cluster
+	clustersMu                sync.RWMutex
+	clusters                  deque.Deque
+	activeCluster             *Cluster
+	activeStateQueue          []bool
+	activeStateQueueInProcess atomic.Bool
 
 	onSendProbe        func(bytesToSend int)
 	onProbeClusterDone func(info ProbeClusterInfo)
+	onActiveChanged    func(isActive bool)
```

Comment:
these are only bound to `StreamAllocator` methods, could these also be an interface?

Reply:
Yeah, it can. Will take a look.